nixos/tests/containers-ipv6.nix

```diff
@@ -0,0 +1,61 @@
+# Test for NixOS' container support.
+
+let
+  hostIp = "fc00::2";
+  localIp = "fc00::1";
+in
+
+import ./make-test.nix ({ pkgs, ...} : {
+  name = "containers-ipv6";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ aristid aszlig eelco chaoflow ];
+  };
+
+  machine =
+    { config, pkgs, ... }:
+    { imports = [ ../modules/installer/cd-dvd/channel.nix ];
+      virtualisation.writableStore = true;
+      virtualisation.memorySize = 768;
+
+      containers.webserver =
+        { privateNetwork = true;
+          hostAddress6 = hostIp;
+          localAddress6 = localIp;
+          config =
+            { services.httpd.enable = true;
+              services.httpd.adminAddr = "foo@example.org";
+              networking.firewall.allowedTCPPorts = [ 80 ];
+              networking.firewall.allowPing = true;
+            };
+        };
+
+      virtualisation.pathsInNixDB = [ pkgs.stdenv ];
+    };
+
+  testScript =
+    ''
+      $machine->waitForUnit("default.target");
+      $machine->succeed("nixos-container list") =~ /webserver/ or die;
+
+      # Start the webserver container.
+      $machine->succeed("nixos-container start webserver");
```

Comment:
This fails for me because there is no no nameserver available:

```
machine# error checking the existence of http://tarballs.nixos.org/sha256/13i04dkd709f0p5f2413sf2y9321pfi4y85ynf8wih6ryphnbk9x:
machine# curl: (6) name lookup timed out
machine# 
machine# trying http://www.docbook.org/xml/5.0/docbook-5.0.zip
machine# Warning: Transient problem: timeout Will retry in 1 seconds. 3 retries left.
machine# Warning: Transient problem: timeout Will retry in 2 seconds. 2 retries left.
machine# Warning: Transient problem: timeout Will retry in 4 seconds. 1 retries left.
machine# curl: (6) Couldn't resolve host 'www.docbook.org'
machine# error: cannot download docbook-5.0.zip from any mirror
machine# builder for ‘/nix/store/mh8hylyfkrmbhjbpz4xkz8f8lx3hcbvc-docbook-5.0.zip.drv’ failed with exit code 1
```